commit/merkleroot/observation.go

```diff
@@ -338,6 +358,28 @@ func (o ObserverImpl) computeMerkleRoot(ctx context.Context, msgs []cciptypes.Me
 	return root, nil
 }
 
+func (o ObserverImpl) ObserveRMNRemoteCfg(
+	ctx context.Context,
+	dstChain cciptypes.ChainSelector) rmntypes.RMNRemoteConfig {
+	// check if dstChain is supported
+	supportsDestChain, err := o.chainSupport.SupportsDestChain(o.nodeID)
+	if err != nil {
+		o.lggr.Warnw("call to SupportsDestChain failed", "err", err)
+		return rmntypes.RMNRemoteConfig{}
+	}
+
+	if !supportsDestChain {
+		return rmntypes.RMNRemoteConfig{}
+	}
+
+	rmnRemoteCfg, err := o.ccipReader.GetRMNRemoteConfig(ctx, dstChain)
+	if err != nil {
+		o.lggr.Warnw("call to GetRMNRemoteConfig failed", "err", err)
+		return rmntypes.RMNRemoteConfig{}
+	}
+	return rmnRemoteCfg
```

Comment:
This is a nit, but I think all of the "do I support this chain" logic is already done in the CCIP reader. I think this code can just be simplified to call the CCIP reader and checking `ErrContractReaderNotFound` or whatever the error is.

```golang
rmnRemoteCfg, err := o.ccipReader.GetRMNRemoteConfig(ctx, dstChain)
if err != nil {
  if errors.Is(err, reader.ErrContractReaderNotFound) {
    // don't support dest chain
    return rmntypes.RMNRemoteConfig{}
  }
  // legitimate error, we should log it.
  o.lggr.Warnw("call to GetRMNRemoteConfig failed", "err", err)
  return rmntypes.RMNRemoteConfig{}
}

return rmnRemoteCfg
```

Reply:
Nice one